### PR TITLE
[BAHIR-173]Update Flink version to 1.5.1

### PR DESCRIPTION
**Pull request by Wosin**
This PR updates the version of Flink to the last official release 1.5.1. 
This update required updating akka version in `flink-connector-akka` module.
`MultiException` was removed since it was imported directly from Flink and is not present in Flink 1.5.1.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,19 +32,19 @@ matrix:
   include:
     - jdk: oraclejdk8
       env:
-        - FLINK_VERSION="1.3.0" SCALA_VERSION="2.11"
+        - FLINK_VERSION="1.5.1" SCALA_VERSION="2.11"
         - CACHE_NAME=JDK8_F130_A
     - jdk: oraclejdk8
       env:
-        - FLINK_VERSION="1.3.0" SCALA_VERSION="2.10"
+        - FLINK_VERSION="1.5.1" SCALA_VERSION="2.10"
```
Comment (eskabetxe) [inline .travis.yml]: the version on scala 2.10 was deprecated on flink version 1.4.0
you have to remove this test

File: .travis.yml [hunk 2]
```diff
@@ -32,19 +32,19 @@ matrix:
...
         - CACHE_NAME=JDK8_F130_B
     - jdk: openjdk8
       env:
-        - FLINK_VERSION="1.3.0" SCALA_VERSION="2.11"
+        - FLINK_VERSION="1.5.1" SCALA_VERSION="2.11"
```
Comment (eskabetxe) [inline .travis.yml]: the version on scala 2.10 was deprecated on flink version 1.4.0
you have to remove this test

Comment (Wosin) [reply .travis.yml]: I know:) I am switching this to scala 2.12

Comment (lresende) [reply .travis.yml]: If scala 2.10 is deprecated, I would remove them from the travis profiles (as they are the ones failing) and would submit support for 2.12 as a new PR.

Comment (Wosin) [reply .travis.yml]: Ok, this seems reasonable. I will remove them completely then.

**Comment by lresende** [comment]
LGTM, any other comments?

**Comment by lresende** [comment]
@eskabetxe do you have a jira account so that I can assign the jira to yourself?

**Pull request closed without merging**